modules/topology_hiding/topo_hiding_logic.c

```diff
@@ -408,6 +429,8 @@ static int topo_dlg_replace_contact(struct sip_msg* msg, struct dlg_cell* dlg)
 			} else {
 				ct_username = ctu.user.s;
 				ct_username_len = ctu.user.len;
+				ct_type = ctu.type;
+				if (ct_type == SIPS_URI_T) prefix_len += 1;
```

Comment:
If RURI and Top RR are `sip:`, meanwhile the Contact header is `sips:`, then we should send out `sip:` in the Contact header if I understand correctly (Contact header only matter when RR is empty). In that case, this part may be broken, as it will calculate with the length of `sips:`, but later, it will send out `sip:` with it. I have to rethink this part.